src/app/components/search-facet/search-facet.component.ts

```diff
@@ -0,0 +1,81 @@
+import {Component, OnInit} from '@angular/core';
+import {SearchFacetType} from '../../../../projects/onto-search/src/lib/onto-search-facet/models/search-facet-type';
+import {SearchFacetGroupModel} from '../../../../projects/onto-search/src/lib/onto-search-facet/models/search-facet-group-model';
+import {SearchFacetModel} from '../../../../projects/onto-search/src/lib/onto-search-facet/models/search-facet-model';
+
+@Component({
+  selector: 'app-search-facet',
+  templateUrl: './search-facet.component.html',
+  styleUrls: ['./search-facet.component.scss']
+})
+export class SearchFacetComponent implements OnInit {
+  public data: SearchFacetGroupModel;
+  public type: SearchFacetType;
+  public apiGroupResponse: any;
+  public apiGroupNameResponse: any;
+  public apiSelectedResponse: any;
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+    this.data = {facetGroupName: '', facetGroup: null, selected: []};
```

Comment:
IMO  initialize it after the api responses and directly set the values.

Reply:
done